### PR TITLE
add lint and test stages for Linux Docker images

### PR DESCRIPTION
**Pull request by MarckK**
INFRA-2879

Working on the above issue, and as a first step we have set up lint and test stages.

- Lint is using Hadolint, without failing the builds (forced pass)
- Test is using Container Structure Tests with tar driver, not `docker run` commands. Basic tests passing.
- Nested directories for storing Dockerfiles now supported.

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -12,7 +12,12 @@ pipeline {
     }
 
     stages { 
-        stage('Build Only') {
+        stage('Lint') {
+            steps {
+                sh "make lint"
+            }
+        }
+        stage('Build and Test') {
```
Comment (slide) [inline Jenkinsfile:20]: I think we would only need to lint when `infra.isTrusted()` is false, e.g., for builds on ci.jenkins.io for PR's and so forth. The builds on trusted only occur on the master branch and we don't merge directly to master.

**Review by slide — CHANGES_REQUESTED** [review]

File: Jenkinsfile (hunk 1 above)
Comment (dduportal) [reply Jenkinsfile:20]: The idea that is tested in this draft PR would be to use the tar export of the image , after the tests. This should be then stashed, and unstashed by the untrusted infra for the "push only".

OR we can keep the current behavior that rebuild the image on the trusted infra, without the lint and tests steps. but it feels weird.

Comment (dduportal) [reply Jenkinsfile:20]: Hello @slide ! We are not sure to understand how it works:

- For the trusted infra, what are the expected steps to be run? Push only (so the tested image should be unstashed) or build and push (and rebuild the image a 2nd time)?
- Is there a reason to not use 2 separated pipelines for both instances? (or a the Jenkinfile for untrusted, and a pipeline library for better control on the untrusted infra)

Comment (dduportal) [reply Jenkinsfile:15]: Feedback applied 👍

**Review by slide — APPROVED** [review]

**Pull request merged**